server/shared/src/main/scala/org/http4s/server/middleware/ErrorHandling.scala

```diff
@@ -42,4 +44,53 @@ object ErrorHandling {
 
   def httpApp[F[_]: MonadThrow](httpApp: HttpApp[F]): HttpApp[F] =
     apply(httpApp)
+
+  object Custom {
+    def handleErrorWith[F[_]: MonadThrow, G[_], A](
+        pf: PartialFunction[Throwable, F[Response[G]]],
+        http: Kleisli[F, A, Response[G]],
```

Comment:
It would be handy, if we will have a contract like this:
```scala
def handleErrorWith[F[_]: MonadThrow, G[_], A](
  http: Kleisli[F, A, Response[G]])(pf: PartialFunction[Throwable, F[Response[G]]])
): Kleisli[F, A, Response[G]]
```
Then one may it use it like:
```scala
Custom.handleErrorWith(http) {
  case _ => ???
}

Reply:
Adjusted for the better ergonomics here.